agent/wsclient/client.go

```diff
@@ -127,23 +130,41 @@ func (cs *ClientServerImpl) Connect() error {
 		return err
 	}
 
+	// gorilla/websocket expects the websocket scheme (ws[s]://)
+	parsedURL.Scheme = "wss"
+
 	// NewRequest never returns an error if the url parses and we just verified
 	// it did above
-	request, _ := http.NewRequest("GET", cs.URL, nil)
+	request, _ := http.NewRequest("GET", parsedURL.String(), nil)
+
 	// Sign the request; we'll send its headers via the websocket client which includes the signature
 	utils.SignHTTPRequest(request, cs.Region, ServiceName, cs.CredentialProvider, nil)
 
-	wsConn, err := cs.websocketConn(parsedURL, request)
-	if err != nil {
-		return err
+	timeoutDialer := &net.Dialer{Timeout: wsConnectTimeout}
+	tlsConfig := &tls.Config{ServerName: parsedURL.Host, InsecureSkipVerify: cs.AcceptInvalidCert}
+
+	//Ensure that NO_PROXY gets set
+	noProxy := os.Getenv("NO_PROXY")
+	dockerHost := os.Getenv("DOCKER_HOST")
```

Comment:
You should use `cfg.DockerEndpoint` (and strip the scheme from it) as that will resolve to the correct endpoint (that's what we use elsewhere).  If you don't want to pass in the whole `cfg` object, you could just pass that single field in.  Or you could simplify the existing code a bit by passing in `cfg` and then extracting the Docker endpoint, region, etc from it.